tests/unit_tests/test_tethys_apps/test_base/test_app_base.py

```diff
@@ -648,12 +648,12 @@ def test_set_custom_setting_object_not_exist(self, mock_app, mock_tas_dne):
     @mock.patch('tethys_apps.models.TethysApp')
     def test_set_custom_setting_type_not_match(self, mock_app):
         setting_name = 'fake_setting'
-        mock_app.objects.get().custom_settings.get.return_value = mock.MagicMock(type='STRING')
+        mock_app.objects.get().custom_settings.get.return_value = mock.MagicMock(type='UUID')
```

Comment:
Please add a new test to test the UUID type_match functionality instead of repurposing an existing test. I'd like to get away from doing the minimal to reach 100% coverage.

Reply:
https://github.com/msouff/tethys/blob/custom_settings_updates/tethys_apps/base/app_base.py#L957-L976

I picked the STRING type previously because it was the first in order, but what actually needs to be tested here is that the 'type_matches' variable is False and that the exception in line 976 is raised. I thought changing the type to UUID was a clever way to test this while hitting line 970 in the process since we still want to hit every line.

Testing the exception at line 970 separately doesn't make sense because it is just passing. The separate test would be unrealistic (like, I'd side_effect the exception and then test that it is throwing that exception?). Let me know what you think.